datm/cime_config/buildnml

```diff
@@ -78,20 +83,28 @@ def _create_namelists(case, confdir, inst_string, infile, nmlgen, data_list_path
     namelist_file = os.path.join(confdir, "datm_in")
     nmlgen.write_output_file(namelist_file, data_list_path, groups=['datm_nml'])
 
-    # Generate datm.streams.xml
-    streams = nmlgen.get_streams()
+    # Determine streams
+    streamlist = nmlgen.get_streams()
+    print "DEBUG: streamlist is ",streamlist
     if datm_presaero != "none":
-        streams.append("presaero.{}".format(datm_presaero))
+        streamlist.append("presaero.{}".format(datm_presaero))
     if datm_topo != "none":
-        streams.append("topo.{}".format(datm_topo))
+        streamlist.append("topo.{}".format(datm_topo))
     if datm_co2_tseries != "none":
-        streams.append("co2tseries.{}".format(datm_co2_tseries))
+        streamlist.append("co2tseries.{}".format(datm_co2_tseries))
     bias_correct = nmlgen.get_value("bias_correct")
-    streams.append(bias_correct)
+    if bias_correct is not None:
+        streamlist.append(bias_correct)
     anomaly_forcing = nmlgen.get_value("anomaly_forcing")
-    streams += anomaly_forcing
-    stream_path = os.path.join(confdir, "datm.streams"+inst_string+".xml" )
-    nmlgen.create_nuopc_stream_files(config, caseroot, streams, stream_path, data_list_path)
+    if anomaly_forcing[0] is not None:
+        streamlist += anomaly_forcing
+    
+    # Generate datm.streams.xml
+    stream_file = os.path.join(_CDEPS_CONFIG,os.pardir, "datm","cime_config","stream_definition_datm.xml")
+    schema_file = os.path.join(_CDEPS_CONFIG,"streams_v2.0.xsd")
```

Comment:
The schema file path should be provided in config_files.xml this has been the method to provide schema until now. 
also os.path.abspath should be used here.

Reply:
I disagree with this. This means that every time the schema changes a new CIME update needs to be made. By having this in CDEPS it is totally self-contained. I expect that the schema will continue to evolve as we iron things out. This is just implementing a separation of concerns.